internal/plugin/connectors/tcp/mssql/authentication_handshake.go

```diff
@@ -0,0 +1,132 @@
+package mssql
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"sort"
+
+	mssql "github.com/denisenkom/go-mssqldb"
+)
+
+// PerformFakeHandshakeForClient carries out a fake handshake with a client
+// 1. reads the prelogin packet
+// 2. writes the prelogin repsonse declaring NO support for encryption
+// 3. reads the login packet
+// 4. writes the login response as a success. TODO: what about failure ?
+func PerformFakeHandshakeForClient(clientConn net.Conn) error {
+	// using the default packet size of 4096 (see go-mssqldb/conn_str.go)
+	clientBuf := mssql.NewTdsBuffer(4096, clientConn)
+	msg, err := ReadClientPrelogin(clientBuf)
+	if err != nil {
+		return fmt.Errorf("bad prelogin: %s", err)
+	}
+
+	// based on wireshark, the prelogin response from the server is almost identical
+	// to the login message from the client, with minor changes as below:
+	msg[mssql.PreloginTHREADID] = []byte{}
+	msg[mssql.PreloginVERSION] = []byte{0x0e, 0x00, 0x0c, 0xa6, 0x00, 0x00}
+	msg[mssql.PreloginENCRYPTION] = []byte{mssql.EncryptNotSup}
+
+	err = WritePreloginResponse(clientBuf, msg)
+	if err != nil {
+		return fmt.Errorf("bad prelogin response: %s", err)
+	}
+
+	// we actually don't care what the client has to say.
+	// we just need for them to not be blocked
+	err = clientBuf.ReadNextPacket()
+	if err != nil {
+		return fmt.Errorf("bad client login read: %s", err)
+	}
+
+	// created an unencrypted connection to mssql, and just copied the raw bytes
+	// this has loginack, and done tokens.
+	_, err = clientConn.Write([]byte{
+		0x04, 0x01, 0x00, 0x64, 0x00, 0x37, 0x01, 0x00, 0xad, 0x36, 0x00, 0x01, 0x74, 0x00, 0x00, 0x04, 0x16, 0x4d, 0x00, 0x69, 0x00, 0x63, 0x00, 0x72, 0x00, 0x6f, 0x00, 0x73, 0x00, 0x6f, 0x00, 0x66, 0x00, 0x74, 0x00, 0x20, 0x00, 0x53, 0x00, 0x51, 0x00, 0x4c, 0x00, 0x20, 0x00, 0x53, 0x00, 0x65, 0x00, 0x72, 0x00, 0x76, 0x00, 0x65, 0x00, 0x72, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0e, 0x00, 0x0c, 0xa6, 0xe3, 0x13, 0x00, 0x04, 0x04, 0x34, 0x00, 0x30, 0x00, 0x39, 0x00, 0x36, 0x00, 0x04, 0x34, 0x00, 0x30, 0x00, 0x39, 0x00, 0x36, 0x00, 0xfd, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
```

Comment:
Given the amount of magic constant values here, it would be good to add a comment to indicate that this is going to be eventually changed to be done at runtime rather than just hardcoded within.